File: patterns/examples/design-patterns-java/src/main/java/br/com/alura/loja/budget/situation/Approved.java
package br.com.alura.loja.budget.situation;

import java.math.BigDecimal;

import br.com.alura.loja.DomainException;
import br.com.alura.loja.budget.Budget;

public class Approved extends BudgetSituation {

	@Override
	public BigDecimal extraDiscountCalculation(Budget budget) {
		return budget.value().multiply(new BigDecimal("0.02"));
	}

	@Override
	public void complete(Budget budget) throws DomainException {
		budget.setSituation(new Completed());
	}

}

File: patterns/examples/budzet/src/BudgetState/Approved.php
<?php

namespace Budzet\BudgetState;

use Budzet\Budget;

class Approved extends BudgetState
{
  public function calculateExtraDiscount(Budget $budget): float
  {
    return $budget->value * 0.02;
  }

  public function complete(Budget $budget) {
    $budget->currentState = new Completed();
  }
}